pkg/handler/common/machine.go

```diff
@@ -419,6 +424,101 @@ func ListMachineDeploymentMetrics(ctx context.Context, userInfoGetter provider.U
 	return ConvertNodeMetrics(nodeDeploymentNodesMetrics, availableResources)
 }
 
+func PatchMachineDeployment(ctx context.Context, userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, sshKeyProvider provider.SSHKeyProvider, seedsGetter provider.SeedsGetter, projectID, clusterID, machineDeploymentID string, patch json.RawMessage) (interface{}, error) {
+	clusterProvider := ctx.Value(middleware.ClusterProviderContextKey).(provider.ClusterProvider)
+	userInfo, err := userInfoGetter(ctx, "")
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	project, err := common.GetProject(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, projectID, nil)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	cluster, err := GetCluster(ctx, projectProvider, privilegedProjectProvider, userInfoGetter, projectID, clusterID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := common.GetClusterClient(ctx, userInfoGetter, clusterProvider, cluster, projectID)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	// We cannot use machineClient.ClusterV1alpha1().MachineDeployments().Patch() method as we are not exposing
+	// MachineDeployment type directly. API uses NodeDeployment type and we cannot ensure compatibility here.
+	machineDeployment := &clusterv1alpha1.MachineDeployment{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: metav1.NamespaceSystem, Name: machineDeploymentID}, machineDeployment); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	nodeDeployment, err := outputMachineDeployment(machineDeployment)
+	if err != nil {
+		return nil, fmt.Errorf("cannot output existing node deployment: %v", err)
+	}
+
+	nodeDeploymentJSON, err := json.Marshal(nodeDeployment)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode existing node deployment: %v", err)
+	}
+
+	patchedNodeDeploymentJSON, err := jsonpatch.MergePatch(nodeDeploymentJSON, patch)
+	if err != nil {
+		return nil, fmt.Errorf("cannot patch node deployment: %v", err)
+	}
+
+	var patchedNodeDeployment *apiv1.NodeDeployment
+	if err := json.Unmarshal(patchedNodeDeploymentJSON, &patchedNodeDeployment); err != nil {
+		return nil, fmt.Errorf("cannot decode patched cluster: %v", err)
+	}
+
+	//TODO: We need to make the kubelet version configurable but restrict it to versions supported by the control plane
```

Comment:
is this TODO still relevant? From what I see we are checking if the version is compatible below